src/TraceEvent/ETWReloggerTraceEventSource.cs

```diff
@@ -407,7 +407,7 @@ private unsafe void Initialize(TraceEventNativeMethods.EVENT_RECORD* rawData)
                         // first event (real time case).   This is really a problem, as we need that information, but we will go ahead and
                         // try to initialize as best we can. 
 
-                        m_source.pointerSize = sizeof(IntPtr);
+                        m_source.pointerSize = Environment.Is64BitOperatingSystem ? 8 : 4;
```

Comment:
I am not sure what is right here. When an ETL is recorded it can contain x64 and x86 stacks. So the structures in the ETL file should be all dependant on the bitness of the OS and not the process starting the trace. 
It could be that depending on the source.pointersize various assumptions are encoded which will lead to followup errors. 
I can strip the change down to the concrete fix to make realtime tracing work and leave the rest as is.